chain/cmd/bandsv/main.go

```diff
@@ -410,128 +225,83 @@ func handleRequestData(c *gin.Context) {
+}
 
-	ops := proof.GetOps()
-	if ops == nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "proof ops not found"})
+func handleExecute(c *gin.Context) {
+	var req ExecuteRequest
+	if err := c.ShouldBindJSON(&req); err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
 
-	height := uint64(resp.Response.Height) + 1
-	var iavlOpData []byte
-	var multistoreOpData []byte
-	for _, op := range ops {
-		opType := op.GetType()
-		if opType == "iavl:v" {
-			iavlOpData = op.GetData()
-		} else if opType == "multistore" {
-			multistoreOpData = op.GetData()
-		}
-	}
-	if iavlOpData == nil || multistoreOpData == nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "proof was corrupted"})
+	rawParams, err := wasm.SerializeParams(req.Code, []byte(req.Params))
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
-
-	var opiavl iavl.IAVLValueOp
-	if iavlOpData == nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"error": "proof was corrupted"})
+	// Call prepare function
+	prepare, err := wasm.Prepare(req.Code, rawParams)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
-	err = cdc.UnmarshalBinaryLengthPrefixed(iavlOpData, &opiavl)
+	var commands []Command
+	err = json.Unmarshal(prepare, &commands)
 	if err != nil {
-		c.JSON(http.StatusInternalServerError, gin.H{"error": err.Error()})
+		c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
 		return
 	}
 
-	odp := OracleDataProof{}
-	odp.RequestId = requestId
-	odp.Data = resp.Response.GetValue()
-	odp.CodeHash = rwr.CodeHash
-	odp.Params = rwr.ParamsRaw
-	odp.Version = uint64(opiavl.Proof.Leaves[0].Version)
-
-	for i := len(opiavl.Proof.LeftPath) - 1; i >= 0; i-- {
-		path := opiavl.Proof.LeftPath[i]
-		imp := IAVLMerklePath{}
-		imp.SubtreeHeight = uint8(path.Height)
-		imp.SubtreeSize = uint64(path.Size)
-		imp.SubtreeVersion = uint64(path.Version)
-		if len(path.Right) == 0 {
-			imp.SiblingHash = path.Left
-			imp.IsDataOnRight = true
-		} else {
-			imp.SiblingHash = path.Right
-			imp.IsDataOnRight = false
+	var answer []string
+	for _, command := range commands {
+		if !allowedCommands[command.Cmd] {
+			c.JSON(http.StatusBadRequest, gin.H{"error": fmt.Errorf("handleRequest unknown command %s", command.Cmd)})
+			return
+		}
+		dockerCommand := Command{
+			Cmd: "docker",
+			Arguments: append([]string{
+				"run", "--rm", "band-provider",
+				command.Cmd,
+			}, command.Arguments...),
+		}
+		query, err := execWithTimeout(dockerCommand, 10*time.Second)
+		if err != nil {
+			c.JSON(http.StatusBadRequest, gin.H{"error": err.Error()})
+			return
 		}
-		odp.MerklePaths = append(odp.MerklePaths, imp)
+		answer = append(answer, string(query))
 	}
 
-	var opms rootmulti.MultiStoreProofOp
-	err = cdc.UnmarshalBinaryLengthPrefixed(multistoreOpData, &opms)
+	b, _ := json.Marshal(answer)
+
+	rawResult, err := wasm.Execute(req.Code, rawParams, [][]byte{b})
```

Comment:
We force 1 data provider here. So user can't really test aggregation logic. But I think that's ok at this stage.